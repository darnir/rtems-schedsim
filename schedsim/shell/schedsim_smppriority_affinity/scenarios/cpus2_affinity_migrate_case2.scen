#
# Create a second task that is lower priority than
# the first but wants to run on the cpu the first
# task was assigned to.
#
rtems_init 2
task_create smp1 3 
# GOAL: 0x09010001, 0x0a010001
# GOAL:   on CPUs 0-1 respectively
cpus IDLE smp1

task_create smp2 4 -a 2
# GOAL: 0x09010001, 0x0a010001
# GOAL:   on CPUs 0 -1 respectively
cpus IDLE smp1

dispatch
# GOAL: 0x09010001, 0x0a010001
# GOAL:   on CPUs 0 -1 respectively
cpus IDLE smp1
