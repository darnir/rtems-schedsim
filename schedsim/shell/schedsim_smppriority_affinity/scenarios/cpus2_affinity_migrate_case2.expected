Script File               : scenarios/cpus2_affinity_migrate_case2.scen
verbose                   : 1
==> 1: #
==> 2: # Create a second task that is lower priority than
==> 3: # the first but wants to run on the cpu the first
==> 4: # task was assigned to.
==> 5: #
==> 6: rtems_init 2
  CPU 0: Thread Heir 0x09010001 priority=255
  CPU 0: Thread Executing 0x09010001 priority=255
  CPU 1: Thread Heir 0x09010002 priority=255
  CPU 1: Thread Executing 0x09010002 priority=255
==> 7: task_create smp1 3 
Creating task with:  Preemption: yes Timeslicing: no
Task (smp1) created: id=0x0a010001, priority=3
Task (smp1) starting: id=0x0a010001, priority=3
  CPU 1: Thread Heir 0x0a010001 priority=3
  CPU 1: Thread Executing 0x0a010001 priority=3
==> 8: # GOAL: 0x09010001, 0x0a010001
==> 9: # GOAL:   on CPUs 0-1 respectively
==> 10: cpus IDLE smp1
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x09010001 @255 / 0x09010001 @255          false
  CPU 1: 0x0a010001 @  3 / 0x0a010001 @  3          false
=== End of Ready Set of Threads
==> 11: 
==> 12: task_create smp2 4 -a 2
Creating task with:  Preemption: yes Timeslicing: no
Task (smp2) created: id=0x0a010002, priority=4
Task (0x0a010002) Set affinity=0x00000002
Task (smp2) starting: id=0x0a010002, priority=4
==> 13: # GOAL: 0x09010001, 0x0a010001
==> 14: # GOAL:   on CPUs 0 -1 respectively
==> 15: cpus IDLE smp1
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x09010001 @255 / 0x09010001 @255          false
  CPU 1: 0x0a010001 @  3 / 0x0a010001 @  3          false
=== End of Ready Set of Threads
==> 16: 
==> 17: dispatch
==> 18: # GOAL: 0x09010001, 0x0a010001
==> 19: # GOAL:   on CPUs 0 -1 respectively
==> 20: cpus IDLE smp1
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x09010001 @255 / 0x09010001 @255          false
  CPU 1: 0x0a010001 @  3 / 0x0a010001 @  3          false
=== End of Ready Set of Threads
